src/globus_sdk/experimental/globus_app/authorizer_factory.py

```diff
@@ -0,0 +1,198 @@
+from __future__ import annotations
+
+import abc
+
+from globus_sdk import AuthLoginClient, ConfidentialAppAuthClient
+from globus_sdk.authorizers import (
+    AccessTokenAuthorizer,
+    ClientCredentialsAuthorizer,
+    GlobusAuthorizer,
+    RefreshTokenAuthorizer,
+)
+from globus_sdk.experimental.tokenstorage import TokenData
+from globus_sdk.services.auth import OAuthTokenResponse
+
+from ._validating_token_storage import ValidatingTokenStorage
+from .errors import MissingTokensError
+
+
+class AuthorizerFactory(metaclass=abc.ABCMeta):
+    """
+    An ``AuthorizerFactory`` is an interface for getting some class of
+    ``GlobusAuthorizer`` from a ``ValidatingTokenStorage`` that meets the
+    authorization requirements used to initialize the ``ValidatingTokenStorage``.
+
+    An ``AuthorizerFactory`` keeps a cache of authorizer objects that are
+    re-used until its ``store_token_response`` method is called.
+    """
+
+    def __init__(self, token_storage: ValidatingTokenStorage):
+        """
+        :param token_storage: The ``ValidatingTokenStorage`` used
+        for defining and validating the set of authorization requirements that
+        constructed authorizers will meet and accessing underlying token storage
+        """
+        self.token_storage = token_storage
+        self._authorizer_cache: dict[str, GlobusAuthorizer] = {}
+
+    def _get_token_data_or_error(self, resource_server: str) -> TokenData:
+        token_data = self.token_storage.get_token_data(resource_server)
+        if token_data is None:
+            raise MissingTokensError(f"No token data for {resource_server}")
+
+        return token_data
+
+    def store_token_response(self, token_res: OAuthTokenResponse) -> None:
```

Comment:
What are the callers for this method?

I'm noticing that the renewing authorizers bypass it (presumably, so that they don't blow the cache), so I'm guessing this is used as part of login flows.

Can we distinguish the name from the token storage method, so that the two are not likely to be confused?

Reply:
This will probably only be called by a `GlobusApp` after its had a user complete a login flow (or someone trying to do `GlobusApp` like things on their own for some reason)

`store_token_response_and_clear_cache` seems reasonable to me?